internal/wasm/interpreter/interpreter.go

```diff
@@ -547,6 +565,14 @@ func (me *moduleEngine) Call(ctx *wasm.ModuleContext, f *wasm.FunctionInstance,
 	return
 }
 
+// failIfClosed returns a sys.ExitError if moduleEngine.CloseWithExitCode was called.
+func (me *moduleEngine) failIfClosed() error {
+	if closed := atomic.LoadUint64(&me.closed); closed != 0 {
```

Comment:
curious how much does this cost as this is invoked anytime we execute `Call`. Anything notable in bench result?

Reply:
I'll add a bench, sure. though I don't think there's a cheaper way to achieve safety and eventually we need this to allow code sections to eventually release (unless we can do it cheaper in assembly). First, I'll add the bench, though!

Reply:
I changed the code to ensure this is only called once per call, wiith notes why.